### PR TITLE
Make Reader use a dict of handlers

### PR DESCRIPTION
**Pull request by fabeschan**

### CODE REVIEW

File: worker/codalabworker/run_manager.py [hunk 1]
```diff
@@ -120,19 +120,22 @@ def memory_bytes(self):
 
 
 class Reader(object):
+
+    read_handlers = {
```
Comment (bkgoksel) [inline worker/codalabworker/run_manager.py]: `self` is not defined in this context as handler is a class variable. As a result this doesn't work in its current form. I guess one way to make it work would be to build this handle dict in `__init__` and make it an instance variable.

**Review by bkgoksel — CHANGES_REQUESTED** [review]
Good refactor but doesn't work as it is.

File: worker/codalabworker/run_manager.py (hunk 1 above)
Comment (fabeschan) [reply worker/codalabworker/run_manager.py]: good idea

Comment (fabeschan) [reply worker/codalabworker/run_manager.py]: done

**Comment by bkgoksel** [comment]
Looks good! Thanks for this PR!

**Review by bkgoksel — APPROVED** [review]

**Pull request merged**